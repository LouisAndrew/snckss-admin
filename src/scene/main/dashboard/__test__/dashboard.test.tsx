import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'

import Dashboard from '..'
import { Views } from '../..'

describe('Dashboard component within main scene. used to change tabs ', () => {
        const mockFn = jest.fn((view: Views) => {})

        afterEach(cleanup)

        const el: React.ReactElement = <Dashboard changeView={mockFn} />

        it('renders without crashing', () => {
                const div = document.createElement('div')
                ReactDOM.render(el, div)
        })

        it('renders correctly', () => {
                const { getByText } = render(el)
                const tab: HTMLElement = getByText('Create Products')

                fireEvent.click(tab)
                expect(mockFn).toBeCalled()
        })

        it('matches snapshot', () => {
                const tree = renderer.create(el).toJSON()
                expect(tree).toMatchSnapshot()
        })
})
