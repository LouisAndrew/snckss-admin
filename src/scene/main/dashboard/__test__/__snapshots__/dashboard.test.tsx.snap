// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component within main scene. used to change tabs  matches snapshot 1`] = `
<ul
  className="tabs nav flex-column nav-tabs"
>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      onClick={[Function]}
    >
      Create Products
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      onClick={[Function]}
    >
      Show Orders
    </a>
  </li>
</ul>
`;
